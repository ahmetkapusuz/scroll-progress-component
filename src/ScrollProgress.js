import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import useScrollPosition from './useScrollPosition';

styled.input.attrs(props => ({
  type: 'text',
  size: props.small ? 5 : undefined,
}));

const ScrollProgressContainer = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
  height: ${({ height }) => height};
  background-color: ${({ color }) => color};
`;

const ScrollProgress = ({ color = '#2B879E', height = '6px' }) => {
  const [width, setWidth] = useState(0);
  const scrollPosition = useScrollPosition();
  useEffect(() => {
    setWidth(
      (scrollPosition / (document.body.offsetHeight - window.innerHeight)) * 100
    );
  }, [scrollPosition]);

  return (
    <ScrollProgressContainer
      color={color}
      height={height}
      style={{ width: `${width}%` }}
    />
  );
};

export default ScrollProgress;
