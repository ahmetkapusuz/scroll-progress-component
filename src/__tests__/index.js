import React from 'react';
import ScrollProgress from '../index';
import renderer from 'react-test-renderer';

test('Should render successfully', () => {
  const component = renderer.create(<ScrollProgress />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
