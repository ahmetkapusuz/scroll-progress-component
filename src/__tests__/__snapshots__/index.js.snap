// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render successfully 1`] = `
<div
  className="ScrollProgress__ScrollProgressContainer-gc2lzd-0 eSsptH"
  color="#2B879E"
  height="6px"
  style={
    Object {
      "width": "0%",
    }
  }
/>
`;
